From 4e8c767efb1bea03da465fee29641b65da1246eb Mon Sep 17 00:00:00 2001
From: Tomas Mraz <tmraz@fedoraproject.org>
Date: Thu, 16 Nov 2023 18:01:12 +0100
Subject: [PATCH 2/3] Revert "set the return-path hard to "<>""

This reverts commit d5d7db6a42f8c3ae6f6bd1efc767ae3fe97f4734.

This is problematic in some use-cases. It needs to be optional.
---
 configure.ac     | 4 ++--
 src/do_command.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 754b8c9..17fa357 100644
--- a/configure.ac
+++ b/configure.ac
@@ -196,14 +196,14 @@ AC_DEFINE(DEBUGGING,1,[Code will be built with debug info.])
 
 AC_DEFINE(MAILARG,"/usr/sbin/sendmail",[There will be path to sendmail.])
 
-AC_DEFINE(MAILFMT,"%s -FCronDaemon -i -odi -oem -oi -t -f '<>'",
+AC_DEFINE(MAILFMT,"%s -FCronDaemon -i -odi -oem -oi -t -f %s",
 [-i    = don't terminate on "." by itself
 -Fx   = Set full-name of sender
 -odi  = Option Deliverymode Interactive
 -oem  = Option Errors Mailedtosender
 -oi   = Ignore "." alone on a line
 -t    = Get recipient from headers
--f '<>' = Empty envelope sender address
+-f %s = Envelope sender address
 -d = undocumented but common flag.])
 
 AC_DEFINE(SYSLOG,1,[Using syslog for log messages.])
diff --git a/src/do_command.c b/src/do_command.c
index 665e1f0..8226a69 100644
--- a/src/do_command.c
+++ b/src/do_command.c
@@ -471,7 +471,7 @@ static int child_process(entry * e, char **jobenv) {
 				if (MailCmd[0] == '\0') {
 					int len;
 
-					len = snprintf(mailcmd, sizeof mailcmd, MAILFMT, MAILARG);
+					len = snprintf(mailcmd, sizeof mailcmd, MAILFMT, MAILARG, mailfrom);
 					if (len < 0) {
 						fprintf(stderr, "mailcmd snprintf failed\n");
 						(void) _exit(ERROR_EXIT);
-- 
2.52.0

